src/charm.py

```diff
@@ -120,14 +120,17 @@ def _configure(self, _):
         # Restart prometheus only if command line arguments have changed,
         # otherwise just reload its configuration.
         if current_services == new_layer.services:
-            self._prometheus_server.reload_configuration()
-            logger.info("Prometheus configuration reloaded")
+            reloaded = self._prometheus_server.reload_configuration()
+            if not reloaded:
+                self.unit.status = BlockedStatus("Failed to load Prometheus config")
```

Comment:
I would first attempt a service restart, followed by health check. Only then Blocked. For your consideration.

Reply:
The intention is not to restart service when Pebble layer has not changed. If Prometheus fails to reload service without Pebble layer change then almost invariably the cause is a bad config. Service restart will not solve this problem because it will also try to load a bad config.

Reply:
My understanding is slightly different:

For example, if the workload app itself crashed, restarting the service here would be helpful. If the config is indeed invalid, then you'd get the same result anyway.
If you do not validate the config yourself, then you probably shouldn't make assumptions on validity; instead make best (blind) effort to get it running.

If OTOH you were validating the config yourself before passing it to the app, then yes, you should actively minimize downtime.

Reply:
- Validating the config can be feature for a future pull request.
- Should the charm be monitoring if the workload application crashed and then restart the app ?